## 4. Train-Validation Split

### 4.1 Feature and Target Separation

In [1]:
# Import libraries
import pandas as pd

# Import libraries for model
from sklearn.model_selection import train_test_split

preprocessed_data = pd.read_csv('data/preprocessed_data.csv')

In [2]:
X = preprocessed_data.drop(columns = ['SK_ID_CURR', 'TARGET']) # Добавление данных в Х без SK_ID_CURR и TARGET
y = preprocessed_data['TARGET']

client_ids = preprocessed_data['SK_ID_CURR']

print(X.shape, y.shape)

(307511, 126) (307511,)


At this stage, the dataset is separated into features (X) and the target (y) to prepare the data for baseline model development.

SK_ID_CURR is excluded from the model features because it is a technical client identifier and does not represent meaningful information for predicting repayment difficulties. However, the identifier is stored separately because it may later be useful for linking model predictions back to individual clients.

In [3]:
# Распределение данных
X_train, X_valid, y_train, y_valid = train_test_split(
    X, 
    y,
    test_size = 0.2,
    stratify = y,
    random_state = 42
)

In [4]:
y_train.value_counts(normalize=True)

TARGET
0    0.919271
1    0.080729
Name: proportion, dtype: float64

In [5]:
y_valid.value_counts(normalize=True)


TARGET
0    0.919272
1    0.080728
Name: proportion, dtype: float64

The data was split into training and validation sets using an 80/20 ratio. The validation set will be used to evaluate and compare models during development.

Since the target variable is imbalanced, stratified sampling was used to preserve approximately the same class distribution in both datasets. As a result, the proportion of clients with repayment difficulties remains around 8.07% in both the training and validation samples.